protocol/ricochet_scanner.go

```diff
@@ -19,8 +128,20 @@ func (rps *RicochetProtocolScanner) ScanProtocol(hiddenService string, osc *conf
 		report.RicochetDetected = false
 	} else {
 		osc.LogInfo("Detected possible ricochet instance\n")
-		// TODO: Actual Analysis
-		report.RicochetDetected = true
+		r := new(goricochet.Ricochet)
+		rsc := &RicochetServiceChecker{osc, r, make(chan bool)}
+		r.Init()
+		go r.ProcessMessages(rsc)
```

Comment:
This is never going to terminate. Not a bug in this code, but a general bug in go-ricochet which I need to fix to avoid cluttering OnionScan runs with dead ricochet listeners.